### PR TITLE
[js] Add multi tenant tutorial

### PR DESCRIPTION
**Pull request by FrankEssenberger**
Relates: https://github.com/SAP/cloud-sdk-backlog/issues/864

## What Has Changed?

Add a tutorial to illustrate multi-tenant application.

### CODE REVIEW

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 1]
```diff
@@ -0,0 +1,497 @@
...
+```
+
+</TabItem>
+</Tabs>
+
+<!-- vale on -->
+
+### Deploy the Application
+
+Before you can deploy the application you need to create a service instance for the destination and XSUAA service in your account.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:189]:
```suggestion
Before you can deploy the application, you need to create a service instance for the destination and XSUAA service in your account.
```
[pp] Provider account for clarity

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 2]
```diff
@@ -0,0 +1,497 @@
...
+You find a `xs-sercurity.json` in the `service-config` folder to create the XSUAA instance.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:190]:
```suggestion
There is an `xs-sercurity.json` file in the `service-config` folder to create the XSUAA instance.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 3]
```diff
@@ -0,0 +1,497 @@
...
+Align the name of your service instances with the ones in the `manifest.yml`.
+Also adjust the route to use the region of your CF e.g. `cfapps.us10.hana.ondemand.com` and also adjust the route path to make it unique in the region.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:192]:
```suggestion
Also, adjust the route to use the region of your CF, e.g., `cfapps.us10.hana.ondemand.com`, and the route path to make it unique in the region.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 4]
```diff
@@ -0,0 +1,497 @@
...
+
+Now you log into CF using the CLI `cf login` and enter the account information of the provider account.
+Navigate to the `multi-tenant-app` folder and execute `cf push` to deploy the application.
+
+### Call the Service
+
+In our example the service is reachable via (for you the URL will be different depending on landscape):
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:199]:
```suggestion
In our example, the service is reachable via (for you the URL will be different depending on landscape):
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 5]
```diff
@@ -0,0 +1,497 @@
...
+
+```
+GET https://multi-tenant-app.cfapps.YOUR_REGION.hana.ondemand.com/service
+```
+
+Let's investigate the implementation of `service-endpoint.ts` above the service will try to obtain a JSON web token (JWT) from the request.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:205]: This sentence reads a bit weird. Should the sentences be split?

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 6]
```diff
@@ -0,0 +1,497 @@
...
+Then it will fetch a destination called `myDestination` using the destination service.
+Since there is no destination with that name, the service will return `404`.
+
+Create a destination with the name in your provider account and also enter some description for that destination.
+After the destination is created the service should return:
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:210]:
```suggestion
After the destination is created, the service should return:
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 7]
```diff
@@ -0,0 +1,497 @@
...
+
+```
+You are on tenant: No jwt given - provider tenant. The destination description is: Provider Destination Description
+```
+
+## The Approuter
+
+The response shows, that there is no JWT attached to the request.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:218]:
```suggestion
The response shows that there is no JWT attached to the request.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 8]
```diff
@@ -0,0 +1,497 @@
...
+This task is done by the application router, the XSUAA and the identity provider (IdP).
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:219]:
```suggestion
This task is done by the application router, the XSUAA, and the identity provider (IdP).
```

approuter is mentioned more times than application router, I would use a single term.

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 9]
```diff
@@ -0,0 +1,497 @@
...
+The details would be too much for this guide and you may have a look at [our approuter guide](../guides/how-to-use-the-approuter.mdx).
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:220]: [nth] shorten then sentence and put it as a note

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 10]
```diff
@@ -0,0 +1,497 @@
...
+Just think of the approuter as an application taking requests and initiating the authorization flow with the XSUAA and IdP.
+Once the user entered the credentials, the request is sent to the target with the JWT issued for the user and account.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:222]:
```suggestion
Once the user enters their credentials, the request is sent to the target with the JWT issued for the user and account.
```

user and account?

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 11]
```diff
@@ -0,0 +1,497 @@
...
+In a productive case the approuter may redirect requests to multiple applications.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:223]:
```suggestion
In a productive case, the approuter may redirect requests to multiple applications.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 12]
```diff
@@ -0,0 +1,497 @@
...
+In our simple example there is just one route.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:224]:
```suggestion
In our simple example, there is just one route.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 13]
```diff
@@ -0,0 +1,497 @@
...
+  - To host two applications (sample application and approuter)
+  - To create a service instance for the destination and XSUAA service
+- You need a basic understanding of SAP BTP and the command line tools to deploy applications.
+
+## The Application
+
+The application is a minimal example which contains only one endpoint containing business logic.
+This endpoint will call the destination service using the SAP Cloud SDK.
+Since the destination service is tenant aware it can be used to illustrate service usage within your multi-tenant application.
+You find the application code in the `multi-tenant-app` folder of the [sample repository](https://github.com/SAP-samples/cloud-sdk-js/tree/main/samples/multi-tenant-application).
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:70]: [req] We mention already at the start of the guide that the code resides here. So we can shorten this to 'You find the application code in the `multi-tenant-app` folder' and then link the exact folder.

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 14]
```diff
@@ -0,0 +1,497 @@
...
+The response shows, that there is no JWT attached to the request.
+This task is done by the application router, the XSUAA and the identity provider (IdP).
+The details would be too much for this guide and you may have a look at [our approuter guide](../guides/how-to-use-the-approuter.mdx).
+Just think of the approuter as an application taking requests and initiating the authorization flow with the XSUAA and IdP.
+Once the user entered the credentials, the request is sent to the target with the JWT issued for the user and account.
+In a productive case the approuter may redirect requests to multiple applications.
+In our simple example there is just one route.
+
+The approuter does not require code, only configuration.
+You find all files in `approuter` folder of the [samples repository](https://github.com/SAP-samples/cloud-sdk-js/tree/main/samples/multi-tenant-application):
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:227]: [req] Same as for the multi-tenant folder, link the approuter folder and remove '...of the samples repository'

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 15]
```diff
@@ -0,0 +1,497 @@
...
+
+- The `manifest.yml` contains the config for the approuter application
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:229]:
```suggestion
- The `manifest.yml` file contains the config for the approuter
```

Some line items end with a period and some don't. Please align lists

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 16]
```diff
@@ -0,0 +1,497 @@
...
+- The `xs-app.json` the config for the route resolution.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:230]:
```suggestion
- The `xs-app.json` file contains the config for the route resolution.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 17]
```diff
@@ -0,0 +1,497 @@
...
+<TabItem value="xs-app.json" className="code-block-height-js thin-scrollbar">
+
+</TabItem>
+</Tabs>
+
+<!-- vale on -->
+
+### Deploy the Approuter
+
+Please adjust the route property in the `manifest.yml`.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:248]:
```suggestion
Please adjust the route property in the `manifest.yml` file.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 18]
```diff
@@ -0,0 +1,497 @@
...
+Replace the placeholders for subdomain and region.
+Log into the provider account using `cf login` and then you call `cf push` from the `approuter` directory.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:250]:
```suggestion
Log into the provider account using `cf login` and call `cf push` from the `approuter` directory.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 19]
```diff
@@ -0,0 +1,497 @@
...
+This deploys the approuter.
+Once the application is deployed you will see a second application running next to your `multi-tenant-app`.
+
+When you open the application you see one route created by the manifest:
+
+```
+GET https://route-prefix-YOUR-SUBDOMAIN.cfapps.YOUR_REGION.hana.ondemand.com/
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:257]: One placeholder has a - another a _

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 20]
```diff
@@ -0,0 +1,497 @@
...
+```
+
+When you follow this route you will get redirected to the `welcomeFile` defined in the `xs-app.json`.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:260]:
```suggestion
When you follow this route you, will get redirected to the `welcomeFile` defined in the `xs-app.json`.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 21]
```diff
@@ -0,0 +1,497 @@
...
+
+<!-- vale on -->
+
+### Deploy the Approuter
+
+Please adjust the route property in the `manifest.yml`.
+Replace the placeholders for subdomain and region.
+Log into the provider account using `cf login` and then you call `cf push` from the `approuter` directory.
+This deploys the approuter.
+Once the application is deployed you will see a second application running next to your `multi-tenant-app`.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:252]:
```suggestion
Once the approuter is deployed, you will see it running as a separate application next to your `multi-tenant-app`.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 22]
```diff
@@ -0,0 +1,497 @@
...
+
+When you open the application you see one route created by the manifest:
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:254]: Here application refers to the approuter. There is a line below 'The `index.html` is located in the application.' It might be confusing to the user which is which.

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 23]
```diff
@@ -0,0 +1,497 @@
...
+
+```
+GET https://route-prefix-YOUR-SUBDOMAIN.cfapps.YOUR_REGION.hana.ondemand.com/
+```
+
+When you follow this route you will get redirected to the `welcomeFile` defined in the `xs-app.json`.
+The `index.html` is located in the application.
+How did the routing work:
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:262]:
```suggestion
How does routing work:
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 24]
```diff
@@ -0,0 +1,497 @@
...
+
+- In our simple scenario the `xs-app.json` defines only one route consisting of a `source`, `target` and `destination`.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:264]:
```suggestion
- In our simple scenario, the `xs-app.json` file defines only one route consisting of a `source`, `target`, and `destination`.
```

File: sidebarsDocsJs.js [hunk 25]
```diff
@@ -62,6 +62,7 @@ module.exports = {
       type: 'category',
       label: 'Tutorials',
       items: [
+        'tutorials/multi-tenant-application',
```
Comment (deekshas8) [inline sidebarsDocsJs.js:65]: [req] This should be the third item

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 26]
```diff
@@ -0,0 +1,497 @@
...
+You are on tenant: a89ea924-d9c2-4gaf-84fb-3ffcff7891011. The destination description is: Subscriber Destination.
+```
+
+The destination of the subscriber account is used at runtime, because the call in the `service-endpoint.ts` uses the selection strategy `subscriberFirst`.
+You can change this by using [different selection strategies](https://sap.github.io/cloud-sdk/api/2.9.0/variables/sap_cloud_sdk_connectivity.DestinationSelectionStrategies).
+This enables consumers to maintain their custom destination used within a multi-tenant application.
+The destination from the provider account could be seen as a fallback.
+
+This is only one example of a tenant-aware service.
+You could imagine a database with a tenantId column to store consumer specific configuration.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:490]:
```suggestion
You could imagine a database with a `tenantId` column to store consumer specific configuration.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 27]
```diff
@@ -0,0 +1,497 @@
...
+  This account is under the control of the application developer.
+- The accounts using the application are called `subscriber accounts`.
+  These accounts are controlled by the consumer.
+- Note that the subscriber and provider account need to be in the same global account.
+  In case you want to offer a service across global accounts you may follow the [service broker approach](https://blogs.sap.com/2022/03/11/consumability-across-global-btp-accounts-via-custom-service-brokers/) which has other limitations.
+- A `tenant-aware service` separates the data of different `subscriber accounts` rigorously.
+  If you build a multi-tenant application all services handling account specific data need to be tenant-aware services.
+
+This guide focuses on points where the SAP Cloud SDK team saw the need of a more detailed guide with samples.
+The selection is related to support issues appearing in the past and therefore it is not a complete guide.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:46]: [pp] I think lines 45 and 46 can be shortened to something like, This is not a detailed/comprehensive guide on the multi-tenancy topic in SAP BTP but rather covers only ...' and put inside a NOTE admonition.

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 28]
```diff
@@ -0,0 +1,497 @@
...
+<!-- vale on -->
+
+These two endpoints are the entry point for the SAP BTP platform to:
+
+- Create and delete a subscription to the application
+- Obtain the services used by the application
+
+In our example the application uses the destination service.
+The application has a binding to a service instance, so it is clear that the application may call the service.
+However, the subscribed account does not know anything about the internal details of our application.
```
Comment (jjtang1985) [inline docs-js/tutorials/multi-tenant-application.mdx:403]:
```suggestion
However, the subscriber account does not know anything about the internal details of our application.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 29]
```diff
@@ -0,0 +1,497 @@
...
+Therefore, the dependencies-endpoint provides the information that the destination service may be used also from the subscribed account.
```
Comment (jjtang1985) [inline docs-js/tutorials/multi-tenant-application.mdx:404]:
```suggestion
Therefore, the `/dependencies` endpoint provides the information that the destination service may be used also from the subscriber account.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 30]
```diff
@@ -0,0 +1,497 @@
...
+
+The code discussed in this guide can be found on the [samples repository](https://github.com/SAP-samples/cloud-sdk-js/tree/main/samples/multi-tenant-application).
+The idea of this tutorial is to explain the main concepts of a multi-tenant applications and how you implement them on SAP BTP.
+The code samples are not a copy-paste solution for productive use, but a didactic sample.
+You need to adjust things like application names or routes for your use case.
+
+The term `multi-tenant application` is used by SAP for applications which are used by different organizations.
+These organizations are potentially different companies or strongly separated parts of one company.
+In any case each organization has its own subaccount on SAP BTP and subscribes to your multi-tenant application.
+The term multi-tenant application is more or less a synonym for a software as a service (SAAS) offering.
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply docs-js/tutorials/multi-tenant-application.mdx:32]: good idea, tried to reorder it. Have a look

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 31]
```diff
@@ -0,0 +1,497 @@
...
+The idea behind this architecture is, that the consumers share the application resources so that they are more effectively used.
+
+Some vocabulary is necessary to understand the next chapters of this tutorial:
+
+- The SAP BTP account in which the application is deployed is called the `provider account`.
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply docs-js/tutorials/multi-tenant-application.mdx:37]: yes that is nice

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 32]
```diff
@@ -0,0 +1,497 @@
...
+The application is a minimal example which contains only one endpoint containing business logic.
+This endpoint will call the destination service using the SAP Cloud SDK.
+Since the destination service is tenant aware it can be used to illustrate service usage within your multi-tenant application.
+You find the application code in the `multi-tenant-app` folder of the [sample repository](https://github.com/SAP-samples/cloud-sdk-js/tree/main/samples/multi-tenant-application).
+The relevant application logic and configuration is located in the following three files:
+
+- In the `applicaiton.ts` the different endpoints are defined.
+  For now, only the `/service` endpoint is relevant which represents our multi-tenant service.
+- In the `manifest.yml` the route to the application is given and the used services are defined.
+- In the `service-endpoint.ts` a small implementation is provided.
```
[the remark this answers is not in the document]
Comment (FrankEssenberger) [reply docs-js/tutorials/multi-tenant-application.mdx:77]: Tried to improve it. Have a look

File: docs-js/tutorials/multi-tenant-application.mdx (hunk 5 above)
Comment (FrankEssenberger) [reply docs-js/tutorials/multi-tenant-application.mdx:206]: Tried to improve it.

File: docs-js/tutorials/multi-tenant-application.mdx (hunk 22 above)
Comment (FrankEssenberger) [reply docs-js/tutorials/multi-tenant-application.mdx:255]: change it to approuter applicaition and added the folder for the applicaiton containing the index as a link

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 33]
```diff
@@ -0,0 +1,506 @@
...
+The application has a binding to a service instance, so it is clear that the application may call the service.
+However, the subscriber account does not know anything about the internal details of our application.
+Therefore, the `/dependencies` endpoint provides the information that the destination service may be used also from the subscriber account.
+
+Remember to add all SAP BTP services used by your application in this list.
+If you do not do that you retrieve a 403 error from the XSUAA when you request a service token on behalf of a subscriber account.
+
+### Creating a Subscription
+
+After you created the instance of the `SAAS provisioning service` in the provider account, you can create a subscription via the user interface.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:420]:
```suggestion
After you have created an instance of the `SaaS provisioning service` in the provider account, you can create a subscription via the user interface.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 34]
```diff
@@ -0,0 +1,506 @@
...
+A subscription is only a route to the provider application including the unique subdomain of subscriber account.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:421]:
```suggestion
A subscription is a route to the provider application, including the unique subdomain of the subscriber account.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 35]
```diff
@@ -0,0 +1,506 @@
...
+The `TENANT_HOST_PATTERN` in the `manifest.yml` of the approuter defines how to extract the subscriber account from the URL.
+A route like:
+
+```
+GET https://route-prefix-YOUR_SUBDOMAIN.cfapps.YOUR_REGION.hana.ondemand.com/service
+```
+
+would mean that the subscriber account is _`YOUR_SUBDOMAIN`_.
+Of course, you want to automate onboarding of new accounts as much as possible.
+Hence, a script in the `subscription-endpoint.ts` creates a route automatically:
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:431]:
```suggestion
To automate the onboarding of new accounts, the script in `subscription-endpoint.ts` does the following:
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 36]
```diff
@@ -0,0 +1,506 @@
...
+
+- It creates a route
+- Binds the created route to the approuter.
+- Returns the route URL so that it can appear in the subscriber account.
+
+The creation of the route uses the [CF API](https://v3-apidocs.cloudfoundry.org/version/3.126.0/index.html).
+Unfortunately, there is no out-of-the-box access of this API when you are in the context of an application.
+The code assume a destination with the name `cf-api` in the sample implementation which contains the access data for the CF API:
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:439]:
```suggestion
The code assumes a destination with the name `cf-api` in the sample implementation which contains the access data for the CF API:
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 37]
```diff
@@ -0,0 +1,506 @@
...
+| authentication    | `OAuth2Password`                                             |
+| user              | a user with permission to the provider account               |
+| password          | password of this user                                        |
+| client id         | `cf`                                                         |
+| client secret     | empty string                                                 |
+| token service URL | `https://login.cf.YOUR_REGION.hana.ondemand.com/oauth/token` |
+| URL               | `https://api.cf.YOUR_REGION.hana.ondemand.com`               |
+
+You have to adjust the URL and token service URL to for your region e.g. `https://api.cf.us10.hana.ondemand.com`.
+Once the destination is present you can subscribe the application and routes are created automatically.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:453]:
```suggestion
Once the destination is present, you can subscribe to the application and routes are created automatically.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 38]
```diff
@@ -0,0 +1,506 @@
...
+
+Log into your second SAP BTP account and create a subscription to the `multi-tenant-app` in the cockpit under **Service->Instances and Subscriptions**.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:455]: [req] 'under' is not a recommended word for [UI interaction](https://developers.google.com/style/ui-elements#verbs). Rephrase the sentence instead as:

Log into your second SAP BTP account. Go to **Service->Instances and Subscriptions** and create a subscription to the `multi-tenant-app`.

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 39]
```diff
@@ -0,0 +1,506 @@
...
+```
+
+you will see a response with the tenant ID from the subscriber account:
+
+```
+You are on tenant: a89ea924-d9c2-4gaf-84fb-3ffcff7891011. The destination description is: Provider Destination Description.
+```
+
+The approuter has extracted the subscriber subdomain from the URL and issued a token for this account.
+As an application developer you can use the token to determine the account which currently calls your code.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:471]:
```suggestion
As an application developer, you can use the token to determine the account which currently calls your code.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 40]
```diff
@@ -0,0 +1,506 @@
...
+
+### Removing a Subscription
+
+If the consumer deletes the subscription, the SAP BTP will invoke the `DELETE` method on the subscription-endpoint.
+The code will remove the route from the app router and make the application not reachable anymore for that user.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:476]:
```suggestion
The code will remove the route from the approuter and make the application unreachable for that consumer.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 41]
```diff
@@ -0,0 +1,506 @@
...
+The details of the implementation can be found in the `subscription-endpoint.ts` of the sample application.
+
+## Real World View
+
+The presented example is totally artificial.
+This chapter tries to elaborate a bit more how an actual multi-tenant application would look like and how the SAP Cloud SDK helps you.
+Different consumer are divided by their unique application URL including their subdomain.
+However, up to know nothing subscriber specific is happening in the implementation.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:484]:
```suggestion
However, up to now, nothing subscriber-specific is happening in the implementation.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 42]
```diff
@@ -0,0 +1,506 @@
...
+### Removing a Subscription
+
+If the consumer deletes the subscription, the SAP BTP will invoke the `DELETE` method on the subscription-endpoint.
+The code will remove the route from the app router and make the application not reachable anymore for that user.
+The details of the implementation can be found in the `subscription-endpoint.ts` of the sample application.
+
+## Real World View
+
+The presented example is totally artificial.
+This chapter tries to elaborate a bit more how an actual multi-tenant application would look like and how the SAP Cloud SDK helps you.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:482]:
```suggestion
This chapter elaborate a bit on what an actual multi-tenant application would look like and how the SAP Cloud SDK helps you.
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 43]
```diff
@@ -0,0 +1,506 @@
...
+```
+
+The destination of the subscriber account is used at runtime, because the call in the `service-endpoint.ts` uses the selection strategy `subscriberFirst`.
+You can change this by using [different selection strategies](https://sap.github.io/cloud-sdk/api/2.9.0/variables/sap_cloud_sdk_connectivity.DestinationSelectionStrategies).
+This enables consumers to maintain their custom destination used within a multi-tenant application.
+The destination from the provider account could be seen as a fallback.
+
+This is only one example of a tenant-aware service.
+You could imagine a database with a `tenantId` column to store consumer specific configuration.
+You extract the value from the JWT as shown in the minimal example:
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:500]:
```suggestion
You can extract the value from the JWT as shown in the example:
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 44]
```diff
@@ -0,0 +1,506 @@
...
+You are on tenant: a89ea924-d9c2-4gaf-84fb-3ffcff7891011. The destination description is: Subscriber Destination.
+```
+
+The destination of the subscriber account is used at runtime, because the call in the `service-endpoint.ts` uses the selection strategy `subscriberFirst`.
+You can change this by using [different selection strategies](https://sap.github.io/cloud-sdk/api/2.9.0/variables/sap_cloud_sdk_connectivity.DestinationSelectionStrategies).
+This enables consumers to maintain their custom destination used within a multi-tenant application.
+The destination from the provider account could be seen as a fallback.
+
+This is only one example of a tenant-aware service.
+You could imagine a database with a `tenantId` column to store consumer specific configuration.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:499]:
```suggestion
Imagine a database with a `tenantId` column to store consumer specific configuration.
```

**Review by deekshas8 — CHANGES_REQUESTED** [review]
I noticed a few more things (mainly grammatical or sentence structuring).

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 45]
```diff
@@ -0,0 +1,515 @@
...
+
+which shows that the request contains a JWT issued for the provider account.
+
+## The Subscription
+
+Up to now, you called the application via the provider account.
+In this chapter you learn how to call the service from a different account.
+
+The first thing to do is to create an instance of the `SaaS provisioning service` in your provider account.
+You find the `saas-registry-config.json` in the [samples repository](https://github.com/SAP-samples/cloud-sdk-js/tree/main/samples/cf-multi-tenant-application).
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:312]:
```suggestion
You can find the `saas-registry-config.json` in the [samples repository](https://github.com/SAP-samples/cloud-sdk-js/tree/main/samples/cf-multi-tenant-application).
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 46]
```diff
@@ -0,0 +1,515 @@
...
+A subscription is a route to the provider application, including the unique subdomain of the subscriber account.
+The `TENANT_HOST_PATTERN` in the `manifest.yml` of the approuter defines how to extract the subscriber account from the URL.
+A route like:
+
+```
+GET https://route-prefix-YOUR_SUBDOMAIN.cfapps.YOUR_REGION.hana.ondemand.com/service
+```
+
+would mean that the subscriber account is _`YOUR_SUBDOMAIN`_.
+Of course, you want to automate onboarding of new accounts as much as possible.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:438]:
```suggestion
```

File: docs-js/tutorials/multi-tenant-application.mdx [hunk 47]
```diff
@@ -0,0 +1,515 @@
...
+This deploys the approuter.
+Once the approuter is deployed, you will see it running as a separate application next to your `multi-tenant-app`.
+
+When you open the approuter application you see one route created by the manifest:
+
+```
+GET https://route-prefix-YOUR_SUBDOMAIN.cfapps.YOUR_REGION.hana.ondemand.com/
+```
+
+When you follow this route you, will get redirected to the `welcomeFile` defined in the `xs-app.json`.
```
Comment (deekshas8) [inline docs-js/tutorials/multi-tenant-application.mdx:277]:
```suggestion
When you follow this route, you will get redirected to the `welcomeFile` defined in the `xs-app.json`.
```

**Review by deekshas8 — APPROVED** [review]
LGTM. Feel free to merge after adding the code snippets.
